sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -345,14 +345,16 @@ object CommandUtils extends Logging {
     !path.getName.startsWith(stagingDir) && DataSourceUtils.isDataPath(path)
   }
 
-  def getSizeInBytesFallBackToHdfs(session: SparkSession, path: Path, defaultSize: Long): Long = {
+  def getSizeInBytesFallBackToHdfs(session: SparkSession, table: CatalogTable): Long = {
     try {
       val hadoopConf = session.sessionState.newHadoopConf()
-      path.getFileSystem(hadoopConf).getContentSummary(path).getLength
+      val tablePath = new Path(table.location)
+      val fs: FileSystem = tablePath.getFileSystem(hadoopConf)
+      fs.getContentSummary(tablePath).getLength
     } catch {
       case NonFatal(e) =>
-        logWarning(s"Failed to get table size from hdfs. Using the default size, $defaultSize.", e)
-        defaultSize
+        logWarning(s"Failed to get table size from HDFS. Using the default data size.", e)
```

Comment:
nit: Drops `s` in the head.

Reply:
btw, we don't need to print `defaultSize` anymore?
```
        val defaultSize = session.sessionState.conf.defaultSizeInBytes
        logWarning(s"Failed to get table size from hdfs. Using the default size, $defaultSize.", e)
        defaultSize
```

Reply:
I think we should format it if need print `defaultSize`: `org.apache.spark.util.Utils.bytesToString`.

Reply:
I think we don't need it anymore. #22502 add `defaultSize` because `defaultSize` may be different from `session.sessionState.conf.defaultSizeInBytes`. But now it is always `session.sessionState.conf.defaultSizeInBytes`.